src/librustc/ty/context.rs

```diff
@@ -1997,3 +1999,13 @@ impl<T, R, E> InternIteratorElement<T, R> for Result<T, E> {
         Ok(f(&iter.collect::<Result<AccumulateVec<[_; 8]>, _>>()?))
     }
 }
+
+fn in_scope_traits<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, id: DefId)
```

Comment:
I'm worried about the proliferation of providers, but I think making `trait_map` public to `rustc::ty` would be worse.